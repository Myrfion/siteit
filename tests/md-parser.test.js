import path from 'path';
import { fileURLToPath } from 'url';
import fs from 'fs';
import { describe, test, expect, beforeAll, jest } from '@jest/globals';
import mdParser from '../src/md-parser.js';

jest.mock('fs');
let content;
beforeAll(() => {
  const filename = fileURLToPath(import.meta.url);
  const dirname = path.dirname(filename);
  content = fs.readFileSync(`${dirname}/sample-files/sample.md`, {
    encoding: 'utf8',
  });
});

describe('test mdParser method functionality', () => {
  test('passing valid File should generate valid html', () => {
    const response = mdParser(content);
    expect(response).toMatchSnapshot();
  });
  test('passing no argument to mdParser should match empty string', () => {
    const response = mdParser();
    expect(response).toBe('');
  });
});
