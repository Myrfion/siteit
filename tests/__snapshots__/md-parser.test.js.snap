// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mdParser method functionality passing valid File should generate valid html 1`] = `
"<h2>Links</h2>
<p><a href="http://dev.nodeca.com">link text</a></p>
<p><a href="http://nodeca.github.io/pica/demo/" title="title text!">link with title</a></p>
<p>Autoconverted link https://github.com/nodeca/pica (enable linkify to see)</p>
<h2>Tables</h2>
<table>
<thead>
<tr><th>Option</th><th>Description</th></tr>
</thead>
<tbody>
<tr><td>data</td><td>path to data files to supply the data that will be passed into templates.</td></tr>
<tr><td>engine</td><td>engine to be used for processing templates. Handlebars is the default.</td></tr>
<tr><td>ext</td><td>extension to be used for dest files.</td></tr>
</tbody>
</table>
<h2>Code</h2>
<p>Inline <code>code</code></p>
<p>Indented code</p>
<pre><code>// Some comments
line 1 of code
line 2 of code
line 3 of code
</code></pre>
<h2>Lists</h2>
<p>Unordered</p>
<ul>
<li>Create a list by starting a line with <code>+</code>, <code>-</code>, or <code>*</code></li>
<li>Sub-lists are made by indenting 2 spaces:
<ul>
<li>Marker character change forces new list start:
<ul>
<li>Ac tristique libero volutpat at</li>
</ul>
<ul>
<li>Facilisis in pretium nisl aliquet</li>
</ul>
<ul>
<li>Nulla volutpat aliquam velit</li>
</ul></li>
</ul></li>
<li>Very easy!</li>
</ul>
<p>Ordered</p>
<ol>
<li>Lorem ipsum dolor sit amet</li>
<li>Consectetur adipiscing elit</li>
<li>Integer molestie lorem at massa</li>
</ol>
<ol>
<li>You can use sequential numbers…</li>
<li>…or keep all the numbers as <code>1.</code></li>
</ol>
<h2>Emphasis</h2>
<p><strong>This is bold text</strong></p>
<p><strong>This is bold text</strong></p>
<p><em>This is italic text</em></p>
<p><em>This is italic text</em></p>
<p><del>Deleted text</del></p>
<p>Superscript: 19<sup>th</sup></p>
<p>Subscript: H<sub>2</sub>O</p>
<p><ins>Inserted text</ins></p>
<p><mark>Marked text</mark></p>
<h2>Typographic replacements</h2>
<p>Enable typographer option to see result.</p>
<p>© © ® ® ™ ™ § § ±</p>
<h1>h1 Heading</h1>
<h2>h2 Heading</h2>
<h3>h3 Heading</h3>
<h4>h4 Heading</h4>
<h5>h5 Heading</h5>
<h6>h6 Heading</h6>
<h2>Horizontal Rules</h2>
<hr>
<hr>
<hr>
<p>Block code “fences”</p>
<pre><code><span class="hljs-function"><span class="hljs-title">Sample</span></span> text here...
</code></pre>
<p>Syntax highlighting</p>
<pre><code class="language-js"><span class="hljs-keyword">var</span> foo = <span class="hljs-keyword">function</span> (<span class="hljs-params">bar</span>) {
  <span class="hljs-keyword">return</span> bar++;
};

<span class="hljs-variable language_">console</span>.<span class="hljs-title function_">log</span>(<span class="hljs-title function_">foo</span>(<span class="hljs-number">5</span>));
</code></pre>
"
`;
